cmake/dependencies.cmake

```diff
@@ -1,26 +1,32 @@
 # hunter dependencies
 # https://docs.hunter.sh/en/latest/packages/
 
-# https://docs.hunter.sh/en/latest/packages/pkg/GTest.html
-hunter_add_package(GTest)
-find_package(GTest CONFIG REQUIRED)
-find_package(GMock CONFIG REQUIRED)
+# Append local modules path if Hunter is not enabled
+if (NOT HUNTER_ENABLED)
+  list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+endif()
+
+if (TESTING)
+  # https://docs.hunter.sh/en/latest/packages/pkg/GTest.html
+  hunter_add_package(GTest)
+  find_package(GTest CONFIG REQUIRED)
+  find_package(GMock CONFIG REQUIRED)
+endif()
 
 # https://docs.hunter.sh/en/latest/packages/pkg/Boost.html
 hunter_add_package(Boost COMPONENTS date_time filesystem random system)
 find_package(Boost CONFIG REQUIRED date_time filesystem random system)
 
 # https://docs.hunter.sh/en/latest/packages/pkg/Microsoft.GSL.html
 hunter_add_package(Microsoft.GSL)
-find_package(Microsoft.GSL CONFIG REQUIRED)
```

Comment:
According to the Hunter [docs](https://hunter.readthedocs.io/en/latest/index.html) (or [here](https://hunter.readthedocs.io/en/latest/quick-start/boost-components.html)), `find_package` is still required.

Could you please check if the code below works in your environment?
(let's keep find_package without CONFIG)
```diff
-find_package(Microsoft.GSL CONFIG REQUIRED)
+find_package(Microsoft.GSL REQUIRED)
```

Reply:
I remember trying this line with no luck. CMake has no `FindMicrosoft.GSL.cmake`, but if I add an empty one, it works. The same with cppcodec - we don't use the output `CPPCODEC_FOUND` anywhere, I just added the pkg-config because I noticed cppcodec provided pkg-config files.

Does Hunter do any magic by providing found package info to CMake? If I removed all find_package() calls, I suspect fuhon would still build, does this sound weird?

Reply:
According to [the faq](https://github.com/cpp-pm/hunter/blob/master/docs/faq/why-do-we-need-hunter-add-package.rst), `hunter_add_package(foo)` will download and install `foo` package while `find_package(foo)` should only look up for files in read-only mode.

`find_package` is a CMake builtin function, which loads package settings. Thus, `find_package` looks to be required while even when hunter is not used.

Reply:
Dropped this commit